homeassistant/components/mqtt/discovery.py

```diff
@@ -239,15 +241,17 @@ async def discovery_done(_):
                 hass, MQTT_DISCOVERY_DONE.format(discovery_hash), None
             )
 
-    hass.data.setdefault(DATA_CONFIG_FLOW_LOCK, asyncio.Lock())
-    hass.data[ALREADY_DISCOVERED] = {}
-    hass.data[PENDING_DISCOVERED] = {}
+    if mqtt_data.data_config_flow_lock is None:
+        mqtt_data.data_config_flow_lock = asyncio.Lock()
```

Comment:
Would it make sense to use default-factory for the lock, as already_discovered and pending_discovered also use a default-factory? Or is that an unnecessary overhead?